{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('AGUACACUGGU')"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "#1 Transcribe – Convert the given DNA sequence into an RNA sequence using the transcribe() method.\n",
    "# For example: Seq(\"AGTACACTGGT\").transcribe()\n",
    "Seq(\"AGTACACTGGT\").transcribe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The code transcribes a DNA sequence into RNA by replacing thymine (T) with uracil (U) and prints the resulting RNA sequence."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AGUACACUGGU\n"
     ]
    }
   ],
   "source": [
    "my_seq=Seq(\"AGTACACTGGT\")\n",
    "rna_seq = my_seq.transcribe()\n",
    "my_seq.transcribe()\n",
    "print(rna_seq) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAIVMG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.12/site-packages/Bio/Seq.py:2879: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "#2 Translate – Convert the given RNA sequence into the corresponding protein sequence using the translate() method.\n",
    "my_seq=Seq(\"AUGGCCAUUGUAAUGGGUAG\")\n",
    "protein_seq = my_seq.translate()\n",
    "print(protein_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAIVMG\n"
     ]
    }
   ],
   "source": [
    "# RNA sequence is generated\n",
    "my_seq = Seq(\"AUGGCCAUUGUAAUGGGUAG\")   \n",
    "\n",
    "# We translate (RNA → Protein)\n",
    "protein_seq = my_seq.translate()\n",
    "\n",
    "# We print the result\n",
    "print(protein_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#3 Difference between DNA and RNA – replace the letters T with U in the DNA sequence using the replace() method and print the result."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AUGGCCAUUGUAAUGGGUAG\n"
     ]
    }
   ],
   "source": [
    "# We create the DNA sequence\n",
    "my_seq = Seq(\"ATGGCCATTGTAATGGGTAG\")\n",
    "\n",
    "# We create the RNA sequence by replacing the letters \"T\" with \"U\"\n",
    "rna_seq = str(my_seq).replace(\"T\", \"U\")\n",
    "\n",
    "print(rna_seq)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "My sequence: ACTCGTCAAGTCGCACTGATCAGTAGCTA\n",
      "DNA complement sequence: TGAGCAGTTCAGCGTGACTAGTCATCGAT\n",
      "reverse complement DNA sequence: TAGCTACTGATCAGTGCGACTTGACGAGT\n",
      "RNA complement sequence TGAGCAGTTCAGCGTGACTAGTCATCGAT\n",
      "reverse complement RNA sequence TAGCTACTGATCAGTGCGACTTGACGAGT\n",
      "RNA sequence: ACUCGUCAAGUCGCACUGAUCAGUAGCUA\n",
      "back transcribe sequence RNA into to the DNA ACTCGTCAAGTCGCACTGATCAGTAGCTA\n",
      "Protein sequence: TRQVALISS\n",
      "Protein 9\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "my_seq = Seq(\"ACTCGTCAAGTCGCACTGATCAGTAGCTA\")\n",
    "complement_dna_seq = my_seq.complement()\n",
    "reverse_dna_seq = my_seq.reverse_complement()\n",
    "rna_seq = my_seq.transcribe()\n",
    "back_transcribe_dna_seq = rna_seq.back_transcribe()\n",
    "complement_rna_seq = rna_seq.complement()\n",
    "reverse_complement_rna_seq = rna_seq.reverse_complement()\n",
    "protein_seq = rna_seq.translate()\n",
    "print(\"My sequence:\", my_seq)\n",
    "print(\"DNA complement sequence:\", complement_dna_seq)\n",
    "print(\"reverse complement DNA sequence:\", reverse_dna_seq)\n",
    "print(\"RNA complement sequence\", complement_rna_seq)\n",
    "print(\"reverse complement RNA sequence\", reverse_complement_rna_seq)\n",
    "print(\"RNA sequence:\", rna_seq)\n",
    "print(\"back transcribe sequence RNA into to the DNA\", back_transcribe_dna_seq)\n",
    "print(\"Protein sequence:\", protein_seq)\n",
    "print(\"Protein\", len(protein_seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#4 Convert DNA sequence to amino acids – Convert the given DNA sequence first into RNA and then into a protein sequence."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA Sequence AUGGCCAUUGUAAUGGGUAG\n",
      "Protein Sequence MAIVMG\n"
     ]
    }
   ],
   "source": [
    "# We create DNA sequences\n",
    "my_seq = Seq(\"ATGGCCATTGTAATGGGTAG\")\n",
    "\n",
    "# We convert DNA to RNA (T → U)\n",
    "rna_seq = my_seq.transcribe()\n",
    "\n",
    "# We translate RNA into protein sequence\n",
    "protein_seq = rna_seq.translate()\n",
    "\n",
    "print(\"RNA Sequence\", rna_seq)\n",
    "print(\"Protein Sequence\", protein_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#5 Extracting Data from a FASTA File – Print the IDs and lengths of all sequences in the .fasta file using the SeqIO.parse() method."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: rc_Z78522.1\n",
      "Sequence Length: 83\n",
      "Sequence: Seq('AGTCCACCGGATTCACTCGATCATATATTCTGCTGTCTCAACAATGATCCTTCC...ACG')\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "file_path = r\"/Users/macbookairm2/Documents/GitHub/Bioinformatics/Biopython/Data/ls_orchid.fasta\"  \n",
    "for seq_record in SeqIO.parse(file_path, \"fasta\"):  \n",
    "    print(\"ID:\", seq_record.id)\n",
    "    print(\"Sequence Length:\", len(seq_record))\n",
    "    print(\"Sequence:\", repr(seq_record.seq)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: rc_Z78522.1\n",
      "Sequence Length: 83\n",
      "Sequence: Seq('AGTCCACCGGATTCACTCGATCATATATTCTGCTGTCTCAACAATGATCCTTCC...ACG')\n",
      "----------------------------------------------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "# Enter the path to the FASTA file\n",
    "file_path = r\"/Users/macbookairm2/Documents/GitHub/Bioinformatics/Biopython/Data/ls_orchid.fasta\"\n",
    "\n",
    "# Read FASTA file and print ID, length information\n",
    "for seq_record in SeqIO.parse(file_path, \"fasta\"):\n",
    "    print(\"ID:\", seq_record.id)\n",
    "    print(\"Sequence Length:\", len(seq_record.seq))  # This should be `.seq`\n",
    "    print(\"Sequence:\", repr(seq_record.seq))  # Shows the sequence in full\n",
    "    print(\"-\" * 100)  # Line to separate each sequence "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#6 Annotations from GenBank File – Extract the names of genes and their start-stop positions by reading the .gbk file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: Z78533.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78532.1\n",
      "Sequence Length: 753\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78531.1\n",
      "Sequence Length: 748\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78530.1\n",
      "Sequence Length: 744\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78529.1\n",
      "Sequence Length: 733\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78527.1\n",
      "Sequence Length: 718\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78526.1\n",
      "Sequence Length: 730\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78525.1\n",
      "Sequence Length: 704\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78524.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78523.1\n",
      "Sequence Length: 709\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78522.1\n",
      "Sequence Length: 700\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78521.1\n",
      "Sequence Length: 726\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78520.1\n",
      "Sequence Length: 753\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78519.1\n",
      "Sequence Length: 699\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78518.1\n",
      "Sequence Length: 658\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78517.1\n",
      "Sequence Length: 752\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78516.1\n",
      "Sequence Length: 726\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78515.1\n",
      "Sequence Length: 765\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78514.1\n",
      "Sequence Length: 755\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78513.1\n",
      "Sequence Length: 742\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78512.1\n",
      "Sequence Length: 762\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78511.1\n",
      "Sequence Length: 745\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78510.1\n",
      "Sequence Length: 750\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78509.1\n",
      "Sequence Length: 731\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78508.1\n",
      "Sequence Length: 741\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78507.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78506.1\n",
      "Sequence Length: 727\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78505.1\n",
      "Sequence Length: 711\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78504.1\n",
      "Sequence Length: 743\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78503.1\n",
      "Sequence Length: 727\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78502.1\n",
      "Sequence Length: 757\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78501.1\n",
      "Sequence Length: 770\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78500.1\n",
      "Sequence Length: 767\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78499.1\n",
      "Sequence Length: 759\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78498.1\n",
      "Sequence Length: 750\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78497.1\n",
      "Sequence Length: 788\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78496.1\n",
      "Sequence Length: 774\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78495.1\n",
      "Sequence Length: 789\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78494.1\n",
      "Sequence Length: 688\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78493.1\n",
      "Sequence Length: 719\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78492.1\n",
      "Sequence Length: 743\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78491.1\n",
      "Sequence Length: 737\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78490.1\n",
      "Sequence Length: 728\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78489.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78488.1\n",
      "Sequence Length: 696\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78487.1\n",
      "Sequence Length: 732\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78486.1\n",
      "Sequence Length: 731\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78485.1\n",
      "Sequence Length: 735\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78484.1\n",
      "Sequence Length: 720\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78483.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78482.1\n",
      "Sequence Length: 629\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78481.1\n",
      "Sequence Length: 572\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78480.1\n",
      "Sequence Length: 587\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78479.1\n",
      "Sequence Length: 700\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78478.1\n",
      "Sequence Length: 636\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78477.1\n",
      "Sequence Length: 716\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78476.1\n",
      "Sequence Length: 592\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78475.1\n",
      "Sequence Length: 716\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78474.1\n",
      "Sequence Length: 733\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78473.1\n",
      "Sequence Length: 626\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78472.1\n",
      "Sequence Length: 737\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78471.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78470.1\n",
      "Sequence Length: 574\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78469.1\n",
      "Sequence Length: 594\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78468.1\n",
      "Sequence Length: 610\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78467.1\n",
      "Sequence Length: 730\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78466.1\n",
      "Sequence Length: 641\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78465.1\n",
      "Sequence Length: 702\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78464.1\n",
      "Sequence Length: 733\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78463.1\n",
      "Sequence Length: 738\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78462.1\n",
      "Sequence Length: 736\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78461.1\n",
      "Sequence Length: 732\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78460.1\n",
      "Sequence Length: 745\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78459.1\n",
      "Sequence Length: 744\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78458.1\n",
      "Sequence Length: 738\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78457.1\n",
      "Sequence Length: 739\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78456.1\n",
      "Sequence Length: 740\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78455.1\n",
      "Sequence Length: 745\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78454.1\n",
      "Sequence Length: 695\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78453.1\n",
      "Sequence Length: 745\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78452.1\n",
      "Sequence Length: 743\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78451.1\n",
      "Sequence Length: 730\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78450.1\n",
      "Sequence Length: 706\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78449.1\n",
      "Sequence Length: 744\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78448.1\n",
      "Sequence Length: 742\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78447.1\n",
      "Sequence Length: 694\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78446.1\n",
      "Sequence Length: 712\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78445.1\n",
      "Sequence Length: 715\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78444.1\n",
      "Sequence Length: 688\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78443.1\n",
      "Sequence Length: 784\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78442.1\n",
      "Sequence Length: 721\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78441.1\n",
      "Sequence Length: 703\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78440.1\n",
      "Sequence Length: 744\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n",
      "ID: Z78439.1\n",
      "Sequence Length: 592\n",
      "Gen: 5.8S rRNA, Start: 380, End: 550\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "# Enter the path to the GenBank file\n",
    "file_path = r\"/Users/macbookairm2/Documents/GitHub/Bioinformatics/Biopython/Data/ls_orchid.gbk\"\n",
    "\n",
    "# We read the file and extract the annotation data\n",
    "for seq_record in SeqIO.parse(file_path, \"genbank\"):\n",
    "    print(\"ID:\", seq_record.id)\n",
    "    print(\"Sequence Length:\", len(seq_record.seq))\n",
    "    \n",
    "    # We iterate over each annotation (feature)\n",
    "    for feature in seq_record.features:\n",
    "        if feature.type == \"gene\":  # Take only the \"gene\" annotation type\n",
    "            gene_name = feature.qualifiers.get(\"gene\", [\"Unknown\"])[0]  # Get gene name\n",
    "            start = feature.location.start\n",
    "            end = feature.location.end\n",
    "            print(f\"Gen: {gene_name}, Start: {start}, End: {end}\")\n",
    "\n",
    "    print(\"-\" * 50)  # Line to separate data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#7 Find the Longest Sequence in a FASTA File – Select and print the longest sequence in the .fasta file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID of the longest sequence: rc_Z78522.1\n",
      "Length: 83\n",
      "Sequence: AGTCCACCGGATTCACTCGATCATATATTCTGCTGTCTCAACAATGATCCTTCCGCAGGTTCACCTACGGAAACCTTGTTACG\n",
      "----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "# Enter the path to the FASTA file\n",
    "file_path = r\"/Users/macbookairm2/Documents/GitHub/Bioinformatics/Biopython/Data/ls_orchid.fasta\"\n",
    "\n",
    "# Variables to find the longest sequence\n",
    "longest_seq = None\n",
    "max_length = 0\n",
    "\n",
    "# We read the file and find the longest sequence\n",
    "for seq_record in SeqIO.parse(file_path, \"fasta\"):\n",
    "    seq_length = len(seq_record.seq)\n",
    "    if seq_length > max_length:  \n",
    "        max_length = seq_length\n",
    "        longest_seq = seq_record\n",
    "\n",
    "# We print the longest sequence\n",
    "if longest_seq:\n",
    "    print(\"ID of the longest sequence:\", longest_seq.id)\n",
    "    print(\"Length:\", max_length)\n",
    "    print(\"Sequence:\", longest_seq.seq)\n",
    "    print(\"-\" * 640)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#8 Calculate GC content – ​​Calculate the GC content of a DNA sequence in percentage using the gc_fraction() method."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: biopython in /opt/anaconda3/lib/python3.12/site-packages (1.85)\n",
      "Requirement already satisfied: numpy in /opt/anaconda3/lib/python3.12/site-packages (from biopython) (1.26.4)\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "GC Content: 46.25%\n"
     ]
    }
   ],
   "source": [
    "# Install the required package\n",
    "%pip install biopython\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqUtils import gc_fraction  # Correct function for GC content calculation\n",
    "my_seq = Seq(\"CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGTTGAGACAACAGAATATATGATCGAGTGCCTTGCATCG\")\n",
    "gc_content = gc_fraction(my_seq) * 100  # Calculate GC content as a percentage\n",
    "print(f\"GC Content: {gc_content:.2f}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "#9 Finding a DNA sequence motif – Check if a given DNA sequence contains a specific motif (e.g. \"ATG\") using the find() method."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motif 'ATG' found at index 59 in the sequence.\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "# DNA sequence\n",
    "my_seq = Seq(\"CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGTTGAGACAACAGAATATATGATCGAGTG\")\n",
    "\n",
    "# Motive to search for\n",
    "motif = \"ATG\"\n",
    "\n",
    "# Since the find() method only works on str objects, we convert Seq to str\n",
    "position = str(my_seq).find(motif)\n",
    "\n",
    "# We print the result\n",
    "if position != -1:\n",
    "    print(f\"Motif '{motif}' found at index {position} in the sequence.\")\n",
    "else:\n",
    "    print(f\"Motif '{motif}' not found in the sequence.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#10 Comparing DNA Sequences – Print out the similarities and differences by comparing two different DNA sequences."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Similar parts: [3, 4, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69]\n",
      "Different parts: [0, 1, 2, 5, 6, 18, 37, 38, 39, 40]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "# Two different DNA sequences\n",
    "seq1 = Seq(\"CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGTTGAGACAACAGAATATATGATCGAGTG\")\n",
    "seq2 = Seq(\"ACGAAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGATATCGGTTGAGACAACAGAATATATGATCGAGTG\")\n",
    "\n",
    "# We compare\n",
    "matching_positions = []\n",
    "non_matching_positions = []\n",
    "\n",
    "# We get the length of both sequences (they must be equal)\n",
    "min_len = min(len(seq1), len(seq2))\n",
    "\n",
    "# We compare two sequences element by element\n",
    "for i in range(min_len):\n",
    "    if seq1[i] == seq2[i]:\n",
    "        matching_positions.append(i)\n",
    "    else:\n",
    "        non_matching_positions.append(i)\n",
    "\n",
    "# We print similar and different parts\n",
    "print(f\"Similar parts: {matching_positions}\")\n",
    "print(f\"Different parts: {non_matching_positions}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-G-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATTG-TTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-A-TCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT--GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-AT-CGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCATT-GTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAG-GTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTA-AGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAGTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAA-GTGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGTA-GGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG--A-AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-A--AGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA--AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA--GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACA-AGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAG-TAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAACAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTAGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGT-AGGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC--AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AA-GTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAAC-AAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n",
      "Aligned sequences:\n",
      "Seq1:  -CGTAA-CAAGGTTTCCGT-AGGTGAACCTGCGGAAGGATCAT-TGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Seq2:  ACG-AAGTA-GGTTTCCGTAAG-TGAACCTGCGGAAGGAT-ATCGGTTGAGACAACAGAATATATGATCGAGTG\n",
      "Score:  64.0\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.12/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "from Bio import pairwise2  # Import pairwise2 for sequence alignment\n",
    "\n",
    "# Two different DNA sequences\n",
    "seq1 = Seq(\"CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGTTGAGACAACAGAATATATGATCGAGTG\")\n",
    "seq2 = Seq(\"ACGAAGTAGGTTTCCGTAAGTGAACCTGCGGAAGGATATCGGTTGAGACAACAGAATATATGATCGAGTG\")\n",
    "\n",
    "# To do Alignment\n",
    "alignments = pairwise2.align.globalxx(seq1, seq2)\n",
    "\n",
    "# We print the result\n",
    "for alignment in alignments:\n",
    "    print(\"Aligned sequences:\")\n",
    "    print(\"Seq1: \", alignment[0])\n",
    "    print(\"Seq2: \", alignment[1])\n",
    "    print(\"Score: \", alignment[2])\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#11 Find potential coding regions (regions that start with the ATG start codon and end with the TAA, TAG, or TGA stop codons) in the given DNA sequence."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Potential coding regions found:\n",
      "ATGGCATCGGTGACCAGTATGCTACGATTAAATGCGTAA\n",
      "ATGCTACGATTAAATGCGTAA\n",
      "ATGGATCGATTCAAGTGA\n"
     ]
    }
   ],
   "source": [
    "# DNA sequence\n",
    "my_Seq = Seq(\"ATGGCATCGGTGACCAGTATGCTACGATTAAATGCGTAAGTTGCCATTATGGATCGATTCAAGTGA\")\n",
    "\n",
    "# Codones\n",
    "start_codon = \"ATG\"\n",
    "stop_codons = [\"TAA\", \"TAG\", \"TGA\"]\n",
    "\n",
    "def find_coding_regions(my_Seq):\n",
    "    coding_regions = []\n",
    "    \n",
    "    # Find the Start codones\n",
    "    start_indices = [i for i in range(len(my_Seq) - 2) if my_Seq[i:i+3] == start_codon]\n",
    "\n",
    "    for start_index in start_indices:\n",
    "        possible_stops = []\n",
    "        \n",
    "        # Find stop codons (pick the ones that fit the reading frame)\n",
    "        for stop_codon in stop_codons:\n",
    "            stop_index = my_Seq.find(stop_codon, start_index + 3)\n",
    "            while stop_index != -1:\n",
    "                if (stop_index - start_index) % 3 == 0:  #A multiple of 3?\n",
    "                    possible_stops.append(stop_index)\n",
    "                    break # Get earliest stop codon\n",
    "                stop_index = my_Seq.find(stop_codon, stop_index + 3)  # Look for the next stop codon\n",
    "        \n",
    "        # If a suitable stop codon is found, add the coding region\n",
    "        if possible_stops:\n",
    "            stop_index = min(possible_stops)  # Select the earliest available stop codon\n",
    "            coding_regions.append(my_Seq[start_index:stop_index + 3])\n",
    "\n",
    "    return coding_regions\n",
    "\n",
    "# Find encoding regions\n",
    "coding_regions = find_coding_regions(my_Seq)\n",
    "\n",
    "# Print results\n",
    "print(\"Potential coding regions found:\")\n",
    "for region in coding_regions:\n",
    "    print(region)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#12 Find tandem repeat motifs (i.e. short nucleotide sequences that are repeated several times in a row) in a DNA sequence and determine their number of repetitions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motif 'AT' appears 7 times in the sequence.\n"
     ]
    }
   ],
   "source": [
    "# DNA sequence\n",
    "my_seq = Seq(\"ATATATCGCGCGCAGCTGATATATAGCGCGCGAT\")\n",
    "\n",
    "# We define the tandem motif\n",
    "motif = \"AT\"\n",
    "\n",
    "# We find the number of repetitions of the motif in the sequence\n",
    "count = my_seq.count(motif)\n",
    "\n",
    "# We print the result\n",
    "print(f\"Motif '{motif}' appears {count} times in the sequence.\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
